### PR TITLE
Prepare spykfunc

### LINKED ISSUE #48: Python support/modules needed for Spykfunc
The following modules are needed for Spykunc. I can take care of the ones not yet ported:

- [x] bb5
- [x] hdfs
- [x] jprops
- [x] lazy-property
- [x] pyarrow
- [x] progress
- [x] sparkmanager
- [x] spark-bbp

To have CI work faster, and for a smoother installation, it would be good to have the following packages available via spack:

- [ ] python3

### PR DESCRIPTION
**Pull request by matz-e**
Add spykfunc. Best to not be squashed to make future rebases easier?

Fixes #48.

### CODE REVIEW

File: var/spack/repos/builtin/packages/mvdtool/package.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/mvdtool/package.py:23]: auto-added part below this should be removed

File: var/spack/repos/builtin/packages/py-bb5/package.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-bb5/package.py]: unused?

File: var/spack/repos/builtin/packages/py-h5py/package.py [hunk 3]
```diff
@@ -53,7 +53,7 @@ class PyH5py(PythonPackage):
     depends_on('py-six', type=('build', 'run'))
 
     # Link dependencies
-    depends_on('hdf5@1.8.4:+hl')
+    depends_on('hdf5@1.8.4:+hl~mpi', when='~mpi')
 
     # MPI dependencies
     depends_on('hdf5+mpi', when='+mpi')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-h5py/package.py]: this should be :  `depends_on('hdf5@1.8.4:+hl+mpi', when='+mpi')`

File: var/spack/repos/builtin/packages/py-jprops/package.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+#     spack install py-jprops
+#
+# You can edit this file again by typing:
+#
+#     spack edit py-jprops
+#
+# See the Spack documentation for more information on packaging.
+# If you submit this package back to Spack as a pull request,
+# please first remove this boilerplate and all FIXME comments.
+#
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-jprops/package.py]: same as above (remove template)

File: var/spack/repos/builtin/packages/spykfunc/package.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+import os
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/spykfunc/package.py:25]: unused?

**Review by pramodk — CHANGES_REQUESTED** [review]
Overall looks good to me.

**Review by pramodk — APPROVED** [review]

**Comment by pramodk** [comment]
Read comments on upstream PR and realise obvious things I didn't notice :)

**Comment by matz-e** [comment]
Yes, I should have removed the boilerplates right away :/

I'll address the comments first thing in the morning, and will push another fix commit. I guess we should just squash this in the end, otherwise there's too much commit shuffling involved.

File: var/spack/repos/builtin/packages/py-jprops/package.py [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyJprops(PythonPackage):
+    """FIXME: Put a proper description of your package here."""
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/py-jprops/package.py]: I was trying to push the description `Java properties file parser for Python` but seems like don't have permission to update PR from your branch:

```
remote: Resolving deltas: 100% (4/4), completed with 4 local objects.
To https://github.com/matz-e/spack.git
 ! [remote rejected]     HEAD -> prepare-spykfunc (permission denied)
error: failed to push some refs to 'https://github.com/matz-e/spack.git'
```

**Review by pramodk — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-jprops/package.py (hunk 6 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/py-jprops/package.py]: It's my repository… I'll pick that fix.

**Comment by matz-e** [comment]
@pramodk I consider this a working version that can go into our tree now. All the necessary tidbits have been added to run a cluster-setup like in nix. Further fixes may have to happen, but these can be done post-merge.

**Comment by pramodk** [comment]
Please retest

**Comment by pramodk** [comment]
Please retest

**Review by pramodk — APPROVED** [review]

**Pull request merged**